hivemind/averaging/matchmaking.py

```diff
@@ -480,7 +480,8 @@ async def pop_next_leader(self) -> PeerID:
                 self.peer_id.to_bytes(),
             ):
                 await asyncio.wait(
-                    {self.update_finished.wait(), self.declared_expiration.wait()}, return_when=asyncio.FIRST_COMPLETED
+                    set(map(asyncio.create_task, (self.update_finished.wait(), self.declared_expiration.wait()))),
```

Comment:
```suggestion
                    {asyncio.create_task(self.update_finished.wait()), asyncio.create_task(self.declared_expiration.wait())},
```

That's much easier to read